### PR TITLE
26963 fluentd add ES_HOST/ES_PORT to docker container

### LINKED ISSUE #26963: Configure ES_HOST in fluentd-elasticsearch addons
It would be ideal to be able to configure the `ES_HOST` in the image via an environment variable rather than have hard coded in the config fie.

https://github.com/ajohnstone/kubernetes/blob/master/cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf#L229

### PR DESCRIPTION
**Pull request by ajohnstone**
fixes #26963 - add ES_HOST/ES_PORT to docker container. Refactor conf.d to separate files for maintainability.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/28137)

### CODE REVIEW

**Comment by a-robinson** [comment]
@k8s-bot ok to test

File: cluster/addons/fluentd-elasticsearch/fluentd-es-image/Dockerfile [hunk 1]
```diff
@@ -50,6 +53,8 @@ RUN td-agent-gem install fluent-plugin-kubernetes_metadata_filter fluent-plugin-
 
 # Copy the Fluentd configuration file.
 COPY td-agent.conf /etc/td-agent/td-agent.conf
+COPY run.sh /
+COPY conf.d /etc/td-agent/conf.d/
 
 # Run the Fluentd service.
-ENTRYPOINT ["td-agent"]
+ENTRYPOINT ["bash", "/run.sh"]
```
Comment (jimmidyson) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-image/Dockerfile]: Any reason not to just use `ENTRYPOINT /run.sh`?

**Comment by jimmidyson** [comment]
One nit, but :+1:

**Comment by a-robinson** [comment]
Thanks @ajohnstone, this is great! LGTM, but would you mind squashing your commits down before we merge it?

**Comment by jimmidyson** [comment]
"Boilerplate header is wrong for: /go/src/k8s.io/kubernetes/cluster/addons/fluentd-elasticsearch/fluentd-es-image/run.sh"

**Comment by ajohnstone** [comment]
What's wrong with it exactly?

**Comment by a-robinson** [comment]
See `hack/boilerplate/boilerplate.sh.txt`

**Comment by ajohnstone** [comment]
@a-robinson @jimmidyson added boiler plate :)
Cheers

**Comment by a-robinson** [comment]
@k8s-bot ok to test

**Comment by a-robinson** [comment]
LGTM, thanks again!

**Comment by a-robinson** [comment]
@k8s-bot ok to test

**Pull request closed without merging**